### PR TITLE
Implementing a signup page and unit testing it

### LINKED ISSUE #11: Sign Up page
Create a Signup page in the web application based on the following design reference:

https://www.figma.com/design/ywMImKNdRD9JRZ5Pz9BME6/Sign-Up-Page?node-id=0-1&t=sk5FPwE4anp05dSt-1

The signup page should contain the following fields and behaviors:

Acceptance criteria:

UI Components:
 Header: "Sign Up" should be prominently displayed.
 Username Field: A text field where users enter their desired username.
 Email Field: A text field for entering the user’s email address.
 Password Field: A password input field that hides characters by default.
 Retype Password Field: A password confirmation field that validates against the Password field.
 Login Button: If the user already has an account, provide a button/link to navigate to the Login page.
 Sign in buttons : To make it easy for the users, add Sign in buttons with Google and github.

Tasks:

- [ ] Username Already Exists Check:
When a user submits the form, if the username already exists in the database, show an error message:
"The user already exists, please choose a different username."
- [ ] Password Validation:
The password field should validate the following criteria (as per the Login page's validation):
Must be at least 8 characters long.
Must contain at least one uppercase letter.
Must contain at least one lowercase letter.
Must contain at least one number.
Must contain at least one special character (e.g., @$!%*#?&).
The user should be able to toggle password visibility.
- [ ] Password Matching:
As the user types in the Retype Password field, display a tick mark if the characters match the Password field.
Show feedback during typing indicating whether the passwords match.
- [ ] Success and Error Messages:
 After successful signup, display the following message:
 "Successfully signed up! You can login now."
UI Design Reference:
Refer to the Figma Prototype for design inspiration. Implement the layout and UI to closely match the provided design.

Backend: Node.js, Express (for form submission handling)
Database: MongoDB (for storing user details)

image

### PR DESCRIPTION
**Pull request by oam67**
Fixes #11

**What was changed?**

For this issue, I created a signup page for the web application. The signup page allows the user to enter their username, email, password, and reconfirmation of their password. The signup page ensures that all information submitted is valid. The signup page also includes a signup button to submit the entered information. In addition, the signup page presents the user with options of signing up with Google or GitHub. Once the user has signed up, they will be redirected to the login page. 

**Why was it changed?**

The signup page was created to allow users to register their own accounts and establish unique login credentials. This page is essential for storing user information (such as username, email, and password) in a database, which enables authentication during login. The code addresses this need by providing functionality for users to submit their credentials securely, ensuring they can create individual accounts that can be used for personalized access and future logins.

**How was it changed?**

The signup page was made by creating SignupScreen.js and modifying the LoginScreen.js and App.js. All the code needed for creating the signup page components, submitting the user credentials, and validating the user credentials is encapsulated in SignupScreen.js. LoginScreen.js was modified so that users could press the sign up button and be redirected to the signup page. App.js was modified so that a route could be established for the signup page. Lastly, I created SignupScreen.test.js to unit test the signup page.

**Screenshots that show the changes (if applicable):**
image
image
image

### CODE REVIEW

File: llm_bioinformatics_research/src/LoginScreen/LoginScreen.js [hunk 1]
```diff
@@ -130,7 +129,7 @@ function LoginScreen() {
             </Link>
```
Comment (Omp06) [inline llm_bioinformatics_research/src/LoginScreen/LoginScreen.js:129]: Would it be possible to add the option of entering your username as well so if the user doesn't remember the email they used or vice versa, they are still able to login?

File: llm_bioinformatics_research/src/SignupScreen/SignupScreen.js [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            >
+              Successfully signed up! You will be redirected to the login page in 5 seconds.
+            </Typography>
+          )}
+
+          <Box sx={{ mt: 2, textAlign: 'center' }}>
+            <Button
+              variant="outlined"
+              fullWidth
+              startIcon={<GoogleIcon />}
```
Comment (Omp06) [inline llm_bioinformatics_research/src/SignupScreen/SignupScreen.js:207]: Adding an option to login with Outlook would be beneficial.

File: llm_bioinformatics_research/src/SignupScreen/SignupScreen.js [hunk 3]
```diff
@@ -0,0 +1,247 @@
+import React, { useState, useEffect } from 'react';
+import { TextField, Button, Container, Typography, Box, IconButton, Link } from '@mui/material';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { useNavigate, Link as RouterLink } from 'react-router-dom';
+import GoogleIcon from '@mui/icons-material/Google';
+import GitHubIcon from '@mui/icons-material/GitHub';
+
+function SignUpScreen() {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
```
Comment (Omp06) [inline llm_bioinformatics_research/src/SignupScreen/SignupScreen.js:10]: Adding the name of the account user would make it easier in the future to save the user information in the database

**Review by AjithAkuthota23 — APPROVED** [review]
Everything looks clean and neat. Good structuring and good test coverage.

There are certain things we need to handle, which will be done in further stages

**Pull request merged**